tools/bumper/bumper.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/pkg/errors"
+)
+
+var logger *log.Logger
+
+type inputParams struct {
+	componentsConfigPath string
+	gitToken             string
+}
+
+const (
+	updatePolicyStatic = "static"
+	updatePolicyTagged = "tagged"
+	updatePolicyLatest = "latest"
+)
+
+func main() {
+	logger = initLog()
+	logger.Printf("~~~~~~~~Bumper Script~~~~~~~~")
+
+	inputArgs := inputParams{}
+	initFlags(&inputArgs)
+
+	githubApi, err := newGithubApi(inputArgs.gitToken)
+	if err != nil {
+		exitWithError(errors.Wrap(err, "Failed to create github api instance"))
+	}
+
+	logger.Printf("Parsing %s", inputArgs.componentsConfigPath)
+	componentsConfig, err := parseComponentsYaml(inputArgs.componentsConfigPath)
+	if err != nil {
+		exitWithError(errors.Wrap(err, "Failed to parse components yaml"))
+	}
+
+	for componentName, component := range componentsConfig.Components {
```

Comment:
There is one missing part. Not sure if it should land in this, but I want to discuss it so we know the way forward.

We want to raise one PR per one component's bump right?

And we want to make sure we do not open another PR for a component that is already pending review/tests.

How are you planning to tackle these two points?

Reply:
>We want to raise one PR per one component's bump right?

Yes. for each components that needs bumping we will open a PR with the githubApi.. do you foresee problems with this?

>make sure we do not open another PR for a component that is already pending review/tests

Yes I plan on adding in the next PRs a function called `isPrAlreadyOpened` that will scan all the current open PRs by title (which should be the same since it's automated).

Reply:
> > We want to raise one PR per one component's bump right?
> 
> Yes. for each components that needs bumping we will open a PR with the githubApi.. do you foresee problems with this?
> 

Not really, I just don't see a logic preparing for that here (which is not bad), so I wanted to make sure it is accounted for.

> > make sure we do not open another PR for a component that is already pending review/tests
> 
> Yes I plan on adding in the next PRs a function called `isPrAlreadyOpened` that will scan all the current open PRs by title (which should be the same since it's automated).

Ack

Reply:
> > > We want to raise one PR per one component's bump right?
> > 
> > 
> > Yes. for each components that needs bumping we will open a PR with the githubApi.. do you foresee problems with this?
> 
> Not really, I just don't see a logic preparing for that here (which is not bad), so I wanted to make sure it is accounted for.
> 

Yeah I just prefer to add all high level logic functions, helps me make sure I didn't forget anything..